src/Microsoft.AspNetCore.Mvc.Core/Routing/UrlHelper.cs

```diff
@@ -51,7 +55,7 @@ public virtual string Action(UrlActionContext actionContext)
                 throw new ArgumentNullException(nameof(actionContext));
             }
 
-            var valuesDictionary = new RouteValueDictionary(actionContext.Values);
+            var valuesDictionary = GetValuesDictionary(actionContext);
```

Comment:
Should we do the same of `RouteUrl`? (see line 109). Both of these are commonly used and are pretty similar